notebooks/official/bigquery-ml/bqml-vertexai-model-registry.ipynb

```diff
@@ -0,0 +1,1465 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #1.    while job.running():
</pre><p>Isn't there a synchronous API to call instead of using a while loop?</p><p> </p><p>You should be able to call <code>job.result</code> since <code>QueryJob</code> derives from <code>_AsyncJob</code> which derives from <code>google.api_core.future.polling.PollingFuture</code></p><p>  </p><p>See: https://github.com/googleapis/python-api-core/blob/5b5e77563229687c901d77b5fdecc18168b535e6/google/api_core/future/polling.py#L116&nbsp;&nbsp;</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/842/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Reply:
<p>I got it from one of Andy's notebooks. Since it's not clear how long it takes for job.result to show the result, isn't the loop better?</p><p> </p>

Reply:
It's a hacky way to do it but this is a standard Python Future, so you should just treat it as such:

```
class Future(object, metaclass=abc.ABCMeta):
    # pylint: disable=missing-docstring
    # We inherit the interfaces here from concurrent.futures.

    """Future interface.
    This interface is based on :class:`concurrent.futures.Future`.
    """
```

Reply:
Since it's just a standard Future, you can do all sorts of interesting stuff with it. See https://docs.python.org/3/library/concurrent.futures.html#module-concurrent.futures for details.

Reply:
Thank you! I just changed it to use `job.result()` instead of the `while` loop.